src/main/java/org/commcare/formplayer/screens/FormplayerSyncScreen.java

```diff
@@ -32,38 +34,27 @@ public void init (SessionWrapper sessionWrapper) throws CommCareSessionException
         Entry commandEntry = sessionWrapper.getPlatform().getEntry(command);
         if (commandEntry instanceof RemoteRequestEntry) {
             PostRequest syncPost = ((RemoteRequestEntry)commandEntry).getPostRequest();
-            setUrl(syncPost.getUrl().toString());
+            url = syncPost.getUrl().toString();
             Hashtable<String, String> params = syncPost.getEvaluatedParams(sessionWrapper.getEvaluationContext());
-            UriComponentsBuilder builder = UriComponentsBuilder.newInstance();
-            // hack because HQ isn't accepting the first query param key properly
-            builder.queryParam("buffer", "buffer");
+            queryParams = new LinkedMultiValueMap<String, String>();
             if (asUser != null) {
-                builder.queryParam("commcare_login_as", URLEncoder.encode(asUser));
+
+                queryParams.add("commcare_login_as", asUser);
             }
             for (String key: params.keySet()){
-                builder.queryParam(key, URLEncoder.encode(params.get(key)));
+                queryParams.add(key, params.get(key));
             }
-            builder.build(true);
-            builtQuery = builder.toUriString();
         } else {
             // expected a sync entry; clear session and show vague 'session error' message to user
             throw new RuntimeException("Initialized sync request while not on sync screen");
         }
     }
 
-    public String getBuiltQuery() {
-        return builtQuery;
-    }
-
-    public void setBuiltQuery(String builtQuery) {
-        this.builtQuery = builtQuery;
+    public MultiValueMap<String, String> getQueryParams() {
+        return queryParams;
```

Comment:
let Spring serialize the data into `application/x-www-form-urlencoded` instead of doing it ourselves. We already do this elsewere in the code (https://github.com/dimagi/formplayer/blob/b4901366edc558af0157c29845cac3756cb56af3/src/main/java/org/commcare/formplayer/services/FormattedQuestionsService.java#L40-L44)

See Spring docs: https://docs.spring.io/spring-framework/docs/current/reference/html/integration.html#rest-template-multipart